Sources/ChessKit/Position.swift

```diff
@@ -63,7 +63,25 @@ public struct Position: Equatable {
     private mutating func _toggleSideToMove() {
         sideToMove.toggle()
     }
-
+    
+    /// Checks if in the current position has insufficient material
+    public func hasInsufficientMaterial() -> Bool {
+        let set = self.pieceSet
+        let pawnsRooksQueens = set.pawns | set.rooks | set.queens
+        
+        if pawnsRooksQueens.nonzeroBitCount == 0 {
+            
+            if set.all.nonzeroBitCount <= 3 {
+                return true
+            }
+            
+            return set.knights.nonzeroBitCount == 0
+            && set.bishops.squares.allSatisfy({ $0.color == .dark || $0.color == .light })
```

Comment:
I think `set.bishops.squares.allSatisfy({ $0.color == .dark || $0.color == .light })` will always return `true` since all the bishop squares will be either dark or light.

One way to do this is to create constant bitboards representing the light and dark squares. For example in `Bitboards.swift` you can add:
``` swift
/// Bitboard representing all the light squares on the board.
static let light: Bitboard = 0xAA55AA55AA55AA55
/// Bitboard representing all the dark squares on the board.
static let dark: Bitboard = ~light
```
along with the existing constant bitboards.

Then you can replace this with

```suggestion
            return set.knights == 0
            && ((set.bishops & .light == 0) || (set.bishops & .dark == 0))
```
`set.bishops & .light` equals 0 if all the bishops are on dark squares and vice versa.

Reply:
I just realised my mistake. What I wanted to check was:
`set.bishops.squares.allSatisfy({ $0.color == .dark })` + `set.bishops.squares.allSatisfy({ $0.color == .light })`

Will implement everything as soon as I can. Thank you for the feedback!